### PR TITLE
Jetpack Connect: Display notice when site is already connected

### LINKED ISSUE #13529: Jetpack Connect: detect users trying to connect a connected site, success message
Currently when a user tries to connect an already connected site, we redirect them to the My Plan page with no message. This can be confusing for users. 

Instead we should keep them on the connect or 'add site' screen with a notice. It should be a success notice like this:

image

cc @johnHackworth @annezazu @tyxla @ryelle

### PR DESCRIPTION
**Pull request by tyxla**
This PR fixes #13529, where if you try to connect an already connected Jetpack site, you get redirected to the **My Plan** page without explanation.

This PR suggests that we move the logic for "already connected" case above, so in case user tries to connect an already connected site, a notice with a notice action will appear. Clicking the notice action will actually redirect the user to the **My Plan** page.

In addition, this PR addresses a couple of minor drive-by stuff:

* Pleasing the linter by altering `JetpackConnectNotices` to use an ES6 class.
* Fixing a misusage of `url` / `siteUrl` prop in `JetpackConnectNotices`, altering the component to use a consistent prop for that. Also cleaning up unnecessary usage of that prop within the component.

Side note: generally this implementation is a hybrid between #9503 and what we had before it.

To test:
* Checkout this branch or get it going on calypso.live
* Go to `/jetpack/connect`
* Submit a URL of a site that's already connected.
* Verify you see the notice:

* Click the "Visit" link and verify it leads to the "My Plan" page for that site.
* Test with some other cases and verify notices work like they did before:
  * Non-connected Jetpack site without Jetpack
  * Non-connected Jetpack site with Jetpack deactivated
  * Non-WordPress site
  * A WordPress.com site
  * A wrong, non-existing URL
* Verify connecting a valid Jetpack site still works properly.

### CODE REVIEW

**Comment by sirreal** [comment]
This will reduce #13500 🎉

File: client/signup/jetpack-connect/index.jsx [hunk 1]
```diff
@@ -199,6 +199,10 @@ class JetpackConnectMain extends Component {
 			return false;
 		}
 
+		if ( this.checkProperty( 'isJetpackConnected' ) ) {
```
Comment (sirreal) [inline client/signup/jetpack-connect/index.jsx]: Thinking out loud and this is outside the scope of this PR, but it would be nice to rethink this whole section.

File: client/signup/jetpack-connect/jetpack-connect-notices.jsx [hunk 2]
```diff
@@ -35,81 +37,87 @@ export default React.createClass( {
...
-			noticeValues.text = this.translate( 'Jetpack couldn\'t be found.' );
+			noticeValues.text = translate( 'Jetpack couldn\'t be found.' );
 			return noticeValues;
 		}
 		if ( this.props.noticeType === 'alreadyConnected' ) {
 			noticeValues.status = 'is-success';
 			noticeValues.icon = 'status';
-			noticeValues.text = this.translate( 'This site is already connected!' );
+			noticeValues.text = translate( 'This site is already connected!' );
+			noticeValues.showDismiss = false;
```
Comment (sirreal) [inline client/signup/jetpack-connect/jetpack-connect-notices.jsx]: This already defaults to false, and could be omitted like in other conditions.

File: client/signup/jetpack-connect/index.jsx (hunk 1 above)
Comment (sirreal) [inline client/signup/jetpack-connect/index.jsx]: Is the reordering necessary? When working on #13500 I noticed that these strings which are passed to `JetpackConnectNotices` do not all match. See bb9e1e52b632d5845a0f819b829902faaa95868c

It may be a question of adding `alreadyOwned` to `JetpackConnectNotices`, I'm not sure if there's a condition `isJetpackConnected && ! userOwnsSite`.

File: client/signup/jetpack-connect/jetpack-connect-notices.jsx (hunk 2 above)
Comment (tyxla) [inline client/signup/jetpack-connect/jetpack-connect-notices.jsx]: Done in https://github.com/Automattic/wp-calypso/pull/13599/commits/a49e5f4d7c945d8057587dac709a2d67fa260d04.

File: client/signup/jetpack-connect/index.jsx (hunk 1 above)
Comment (tyxla) [inline client/signup/jetpack-connect/index.jsx]: > Thinking out loud and this is outside the scope of this PR, but it would be nice to rethink this whole section.

:+1: agree, but let's keep it for another PR 😉 

> Is the reordering necessary? When working on #13500 I noticed that these strings which are passed to JetpackConnectNotices do not all match. See bb9e1e5

Yes, because `alreadyOwned` will redirect the user to plans. Although I don't 100% remember, there was some rationale to keep that separate. Maybe @johnHackworth can shed some light on that.

**Comment by tyxla** [comment]
@sirreal it appears we were not checking if the site is connected when checking a particular URL. I've added that in https://github.com/Automattic/wp-calypso/pull/13599/commits/28f11567cbd220df84064c1753e048c5dd2d5b48, can you test again with clean application data?

File: client/signup/jetpack-connect/index.jsx (hunk 1 above)
Comment (roccotripaldi) [inline client/signup/jetpack-connect/index.jsx]: I think something is a bit off here. Running this PR, I get unexpected behavior trying to link an additional user to an already-connected site. I get the notice and I'm not allowed to connect.

File: client/signup/jetpack-connect/jetpack-connect-notices.jsx (hunk 2 above)
Comment (sirreal) [inline client/signup/jetpack-connect/jetpack-connect-notices.jsx]: Oops, I was wrong. It's set to `false`, then immediately changed to `true` in the following conditional block. 🎉  Mutability 🎉 

You can drop a49e5f4, sorry about that.

**Comment by sirreal** [comment]
> I've added that in 28f1156, can you test again with clean application data?

@tyxla This issue is good for me now 👍 

> We need to ensure that we can link additional users to already-connected sites.

Nice catch @roccotripaldi 

1. Connect a Jetpack site
1. Try to connect a new user to the same site
1. ❗️  You get this message, but you haven't connected to the site.

**Comment by tyxla** [comment]
Thanks for the good catches, folks. I thought there might be something not-so-obvious here. Will work on that next week.

**Comment by rickybanister** [comment]
This looks great from my end.

**Comment by tyxla** [comment]
@sirreal @roccotripaldi this one is ready for another review

File: client/signup/jetpack-connect/index.jsx [hunk 3]
```diff
@@ -234,10 +229,16 @@ class JetpackConnectMain extends Component {
 		if ( ! this.checkProperty( 'isJetpackActive' ) ) {
 			return 'notActiveJetpack';
 		}
-		if ( ! this.checkProperty( 'isJetpackConnected' ) ) {
+		if (
+			! this.checkProperty( 'isJetpackConnected' ) ||
+			( this.checkProperty( 'isJetpackConnected' ) && ! this.checkProperty( 'userOwnsSite' ) )
```
Comment (sirreal) [inline client/signup/jetpack-connect/index.jsx]: What is the purpose of this change?

When I connect a site, then try to connect again with a different user, I wind up with the notice for `alreadyConnectedByOtherUser`.

Comment (roccotripaldi) [inline client/signup/jetpack-connect/index.jsx]: Ah yeah, that's a weird one. I'll try to talk it out:

If the site is un-connected, or if the site is connected by someone else, we can set the status to 'notConnectedJetpack'

With 'notConnectedJetpack' status, the current user is able to connect.

Comment (roccotripaldi) [inline client/signup/jetpack-connect/index.jsx]: > When I connect a site, then try to connect again with a different user, I wind up with the notice for alreadyConnectedByOtherUser.

Can you list out the steps to wind up in that state? Did you fully log out, or use an incognito window when trying to link an other user?

Comment (sirreal) [inline client/signup/jetpack-connect/index.jsx]: It was a different browser environment, I'll repro tomorrow and record the exact steps.

Comment (sirreal) [inline client/signup/jetpack-connect/index.jsx]: I've been unable to reproduce the behaviour I described today.

Your English description of this change sounds right to me, so 👍

**Comment by tyxla** [comment]
Thank you for the detailed and thorough review, @sirreal and @roccotripaldi 🙇
